inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -67,7 +67,17 @@ def populate_arxiv_document(obj, eng):
     arxiv_id = get_arxiv_id(obj.data)
     url = current_app.config['ARXIV_PDF_URL'].format(arxiv_id=arxiv_id)
 
-    if not is_pdf_link(url):
+    # Check the first few bytes of the url to ensure it is a PDF file.
+    is_valid_pdf_link = is_pdf_link(url)
+    if not is_valid_pdf_link:
+        # Try with the alternative arxiv url.
+        alternative_url = current_app.config['ARXIV_PDF_URL_ALTERNATIVE']\
+            .format(arxiv_id=arxiv_id)
+        is_valid_pdf_link = is_pdf_link(alternative_url)
+        if is_valid_pdf_link:
+            url = alternative_url
+
+    if not is_valid_pdf_link:
         if NO_PDF_ON_ARXIV in requests.get(url).content:
```

Comment:
Maybe it worth to run this check also on the result of the first request? If the author didn't make the source available then it's probably not worth querying the second URL. Also then you could put this code in a for-loop or similar?

Reply:
Ok I will try with the for loop.

Reply:
Done